lib/queue.rb

```diff
@@ -1,31 +1,73 @@
 class Queue
+  MAX_SIZE = 20
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new
+    @front = -1
+    @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == 0 && @back == MAX_SIZE - 1 || @back == @front - 1 % MAX_SIZE - 1
+      raise ArgumentError
+    elsif @store.empty? 
+      @front = 0
+      @back = 0
+    elsif @back == MAX_SIZE - 1 && @front != 0
+      @back = 0
+    else 
+      @back += 1
+    end
+
+    @store[@back] = element
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    if @store.empty?
+      raise ArgumentError
+    end
+
+    removed_num = @store[@front]
+    @store[@front] = nil
+
+    if @front == @back
+      @front = -1
+      @back = -1
+    elsif @front + 1 == MAX_SIZE
+      @front = 0
+    else
+      @front += 1
+    end
```

Comment:
Just be aware you can do things like:

```suggestion
    else
      @front = (front + 1) % MAX_SIZE
    end
```